File: LLD-Python/solid_principle/single_responsibility/invoice/invoice.py
from tax.tax_calculator import TaxCalculator

class Invoice:
    def __init__(self, amount: float, taxCalculator: TaxCalculator):
        self.amount = amount
        self.taxCalculator = taxCalculator

    def getTotalAmount(self) -> float:
        return self.amount + self.taxCalculator.calculateTax(self.amount)

File: LLD - Java/src/main/java/org/solid/srp/di/Invoice.java
package org.solid.srp.di;

import org.solid.srp.TaxCalculator;

public class Invoice {
    private double amount;
    private TaxCalculator  taxCalculator;

    public Invoice(double amount, TaxCalculator taxCalculator) {
        this.amount = amount;
        this.taxCalculator = taxCalculator;
    }

    public double getAmount() {
        return amount + taxCalculator.calculateTax(amount);
    }
}
